### PR TITLE
1.4 review - GP Synchro corrections (31/03)

### PR DESCRIPTION
**Pull request by ecino**
Improve GP Syncrhonization and code refactoring between modules child_sync and sponsorship_sync.

### CODE REVIEW

File: child_sync_gp/model/child.py [hunk 1]
```diff
@@ -39,77 +39,60 @@ def write(self, cr, uid, ids, vals, context=None):
...
+                                       child.code[:2])
+        file_name = "{0}_{1}.jpg".format(child.code, date_pic)
+        picture_file = TemporaryFile()
+        picture_file.write(base64.b64decode(pictures.fullshot))
+        picture_file.flush()
+        picture_file.seek(0)
+
+        # Upload file to shared folder
+        smb_conn = SMBConnection(smb_user, smb_pass, 'openerp', 'nas')
+        if smb_conn.connect(smb_ip, 139):
```
Comment (CSester) [inline child_sync_gp/model/child.py]: As ip is in the configuration, you could also put the port in.

File: sponsorship_sync_gp/model/child.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def attach_pictures(self, cr, uid, ids, pictures_id, context=None):
+        """ Don't put contract in biennial state if biennial is more
+            recent than cs. """
+        gp_connect = gp_connector.GPConnect()
+        res = super(child_property, self).attach_pictures(
+            cr, uid, ids, pictures_id, context)
+        if not isinstance(ids, list):
+            ids = [ids]
+
+        case_study = self.browse(cr, uid, ids, context)[0]
```
Comment (CSester) [inline sponsorship_sync_gp/model/child.py]:
```python
case_study = self.browse(cr, uid, ids[0], context)
```
Or do you process this way to get the most recent one ?

File: sponsorship_sync_gp/model/gp_connector.py [hunk 3]
```diff
@@ -278,3 +278,48 @@ def delete_contracts(self, ids):
...
+                self.transfer_mapping[child.transfer_country_id.code] \
+                if child.transfer_country_id else 'NULL'
+            update_fields += ", id_motif_fin={0}".format(end_reason)
+            # We don't put a child transfer in ending reason of a sponsorship
+            if not child.transfer_country_id:
+                pole_sql = "UPDATE Poles SET TYPEP = IF(TYPEP = 'C', " \
+                           "'A', 'F'), id_motif_fin={0}, datefin=curdate() " \
+                           "WHERE codespe='{1}' AND TYPEP NOT IN " \
+                           "('F','A')".format(end_reason, child.code)
+                logger.info(pole_sql)
```
Comment (CSester) [inline sponsorship_sync_gp/model/gp_connector.py]: debug message?

File: sponsorship_sync_gp/model/gp_connector.py [hunk 4]
```diff
@@ -278,3 +278,48 @@ def delete_contracts(self, ids):
...
+                self.query(pole_sql)
+
+        if child.state == 'P':
+            # Remove delegation and end_reason, if any was set
+            update_fields += ", datedelegue=NULL, codedelegue=''" \
+                             ", id_motif_fin=NULL"
+
+        sql_query = update_string % (update_fields, child.code)
+        logger.info(sql_query)
```
Comment (CSester) [inline sponsorship_sync_gp/model/gp_connector.py]: Same as 301

File: sponsorship_sync_gp/model/gp_connector.py (hunk 3 above)
Comment (ecino) [inline sponsorship_sync_gp/model/gp_connector.py]: It allows to see in the console of server when requests are made to GP database. I didn't want to remove that.

File: sponsorship_sync_gp/model/child.py (hunk 2 above)
Comment (ecino) [inline sponsorship_sync_gp/model/child.py]: No, it's just because we call it with only one id so your correction is fine.

**Comment by CSester** [comment]
:+1:

**Pull request merged**